pyanaconda/ui/tui/spokes/__init__.py

```diff
@@ -170,15 +166,14 @@ def prompt(self, entry = None):
             return _("Enter new value for '%s' and press enter\n") % entry.title
 
     def input(self, entry, key):
-        if entry.aux.match(key):
-            self.value = key
-            self.close()
-            return True
-        else:
+        (valid, err_msg) = check_name(key)
+        if entry.aux.match(key) and valid:
```

Comment:
Put the boolean value in the first place, it's cheaper to check it than do the ``entry.aux.match(key)`` call. And do we want to call ``check_name()`` before ``entry.aux.match()`` is called? To me it looks like we should first check if the intput is valid then check if it is a valid user name. Or do just one of the checks.